pkg/multicluster/watcher.go

```diff
@@ -0,0 +1,109 @@
+package multicluster
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/go-utils/contextutils"
+	"github.com/solo-io/skv2/pkg/api/kube/core/v1/controller"
+	"github.com/solo-io/skv2/pkg/multicluster/kubeconfig"
+	"github.com/solo-io/skv2/pkg/reconcile"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	LocalCluster = ""
+)
+
+// ClusterHandler is passed to RunClusterWatcher to handle select cluster events.
+type ClusterHandler interface {
```

Comment:
explain this interface is implemented internally and exposed for custom use cases